### PR TITLE
feat: onboard new destination bloomreach

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?

Onboard new destination: `Bloomreach`

## What is the related Linear task?

Resolves INT-1823

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3185/test-report.html

**Review by shrouti1507 — APPROVED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by sandeepdsvs — DISMISSED** [review]

File: test/integrations/destinations/bloomreach/router/data.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  destType,
+  destination,
+  traits,
+  properties,
+  headers,
+  endpoint,
+  RouterInstrumentationErrorStatTags,
+} from '../common';
+
+const routerRequest = {
```
Comment (utsabc) [inline test/integrations/destinations/bloomreach/router/data.ts:13]: Please use the new format of test case writing add description and success criteria

File: src/cdk/v2/destinations/bloomreach/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      const data = .message.().({
+        "customer_ids": customerIDs,
+        "timestamp": $.toUnixTimestamp({{{{$.getGenericPaths("timestamp")}}}}),
+        "properties": .properties,
+        "event_type": $.context.event,
+      });
+
+      $.context.payload = $.removeUndefinedAndNullValues({name: $.CUSTOMER_EVENT_COMMAND, data})
+
+  - name: buildResponse
```
Comment (utsabc) [inline src/cdk/v2/destinations/bloomreach/procWorkflow.yaml:103]: Please add reference to destination docs wherever relevent in description fields

File: test/integrations/destinations/bloomreach/router/data.ts (hunk 1 above)
Comment (Gauravudia) [reply test/integrations/destinations/bloomreach/router/data.ts:13]: Mocking is not supported in new format

Comment (utsabc) [reply test/integrations/destinations/bloomreach/router/data.ts:13]: Add two fields description and success criteria that should be fine

File: src/cdk/v2/destinations/bloomreach/procWorkflow.yaml (hunk 2 above)
Comment (Gauravudia) [reply src/cdk/v2/destinations/bloomreach/procWorkflow.yaml:103]: added doc ref in config.ts

File: test/integrations/destinations/bloomreach/router/data.ts (hunk 1 above)
Comment (Gauravudia) [reply test/integrations/destinations/bloomreach/router/data.ts:13]: this is already added

**Review by utsabc — APPROVED** [review]

**Review by sandeepdsvs — APPROVED** [review]

**Pull request merged**